Sources/Plasma/FeatureLib/pfPython/pyImageLibModGlue.cpp

```diff
@@ -0,0 +1,163 @@
+
+If you modify this Program, or any covered work, by linking or
+combining it with any of RAD Game Tools Bink SDK, Autodesk 3ds Max SDK,
+NVIDIA PhysX SDK, Microsoft DirectX SDK, OpenSSL library, Independent
+JPEG Group JPEG library, Microsoft Windows Media SDK, or Apple QuickTime SDK
+(or a modified version of those libraries),
+containing parts covered by the terms of the Bink SDK EULA, 3ds Max EULA,
+PhysX SDK EULA, DirectX SDK EULA, OpenSSL and SSLeay licenses, IJG
+JPEG Library README, Windows Media SDK EULA, or QuickTime SDK EULA, the
+licensors of this Program grant you additional
+permission to convey the resulting work. Corresponding Source for a
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+
+
+#include <Python.h>
+#include "pyKey.h"
+
+#include "pyImageLibMod.h"
+#include "pyImage.h"
+#include "plGImage/plMipmap.h"
+
+// glue functions
+PYTHON_CLASS_DEFINITION(ptImageLibMod, pyImageLibMod);
+
+PYTHON_DEFAULT_NEW_DEFINITION(ptImageLibMod, pyImageLibMod)
+PYTHON_DEFAULT_DEALLOC_DEFINITION(ptImageLibMod)
+
+PYTHON_INIT_DEFINITION(ptImageLibMod, args, keywords)
+{
+    PyObject* keyObj = nullptr;
+    if (!PyArg_ParseTuple(args, "O", &keyObj))
+    {
+        PyErr_SetString(PyExc_TypeError, "__init__ expects a ptKey");
+        PYTHON_RETURN_INIT_ERROR;
+    }
+    if (!pyKey::Check(keyObj))
+    {
+        PyErr_SetString(PyExc_TypeError, "__init__ expects a ptKey");
+        PYTHON_RETURN_INIT_ERROR;
+    }
+    pyKey* key = pyKey::ConvertFrom(keyObj);
+    self->fThis->setKey(*key);
+    PYTHON_RETURN_INIT_OK;
+}
+
+PYTHON_METHOD_DEFINITION(ptImageLibMod, getImage, args)
+{
+    ST::string name;
+    if (!PyArg_ParseTuple(args, "O&", PyUnicode_STStringConverter, &name)) {
+        PyErr_SetString(PyExc_TypeError, "getImage expects an image name");
+        PYTHON_RETURN_ERROR;
+    }
+
+    pyImage* image = self->fThis->GetImage(name);
+    if (image)
+        return pyImage::New(image->GetKey());
+
+    PYTHON_RETURN_NONE;
+}
+
+PYTHON_METHOD_DEFINITION_NOARGS(ptImageLibMod, getImages)
+{
+    const std::vector<pyImage*> imageList = self->fThis->GetImages();
+    PyObject* retVal = PyTuple_New(imageList.size());
+    for (size_t curKey = 0; curKey < imageList.size(); curKey++)
+        PyTuple_SET_ITEM(retVal, curKey, pyImage::New(imageList[curKey]->GetKey()));
+    return retVal;
+}
+
+PYTHON_METHOD_DEFINITION_NOARGS(ptImageLibMod, getNames)
+{
+    std::vector<ST::string> nameList = self->fThis->GetImageNames();
+    PyObject* retVal = PyTuple_New(nameList.size());
+    for (size_t curKey = 0; curKey < nameList.size(); curKey++)
+        PyTuple_SET_ITEM(retVal, curKey, PyUnicode_FromSTString(nameList[curKey])); // steals the nameList ref
```

Comment:
```suggestion
        PyTuple_SET_ITEM(retVal, curKey, PyUnicode_FromSTString(nameList[curKey]));
```

`PyTuple_SET_ITEM` steals the result of `PyUnicode_FromSTString(nameList[curKey])`